Darwin/StitchCore/StitchCore/Core/Auth/Internal/StitchAuthImpl.swift

```diff
@@ -139,40 +140,105 @@ internal final class StitchAuthImpl: CoreStitchAuth<StitchUserImpl>, StitchAuth
+     */
+    public func switchToUserWithId(_ userId: String,
+                             _ completionHandler: @escaping (StitchResult<StitchUser>) -> Void) {
+        dispatcher.run(withCompletionHandler: completionHandler) {
+            return try self.switchToUserWithIdInternal(userId)
+        }
+    }
+    
+    /**
+     * Removes the current active user from the list of all users
+     * associated with this application. If there is no currently active user, then the
+     * function will return with success.
+     * Additionally, this method will clear all user data including any synchronized databases.
+     *
+     * - parameters:
+     *     - completionHandler: The completion handler to call when the switch is complete.
+     *                          This handler is executed on a non-main global `DispatchQueue`.
+     */
+    public func removeUser(_ completionHandler: @escaping (StitchResult<Void>) -> Void) {
+        dispatcher.run(withCompletionHandler: completionHandler) {
+            return try self.removeUserInternal(withUserId: nil)
+        }
+    }
+    
+    /**
+     * Removes the user with the provided id from the list of all users
+     * associated with this application. If the user was logged in, the user will
+     * be logged out before being removed. The user must exist in the list of all
+     * users who have logged into this application otherwise this will throw a StitchServiveError.
+     * Additionally, this method will clear all user data including any synchronized databases.
+     *
+     * - parameters:
+     *     - userId: A String specifying the desired `userId`
+     *     - completionHandler: The completion handler to call when the switch is complete.
+     *                          This handler is executed on a non-main global `DispatchQueue`.
+     */
+    public func removeUserWithId(userId userId: String,
+                                 _ completionHandler: @escaping (StitchResult<Void>) -> Void) {
         dispatcher.run(withCompletionHandler: completionHandler) {
-            self.logoutInternal()
+            return try self.removeUserInternal(withUserId: userId)
         }
     }
+    
+    /**
+     * Returns a list of all users who have logged into this application, with the exception of
+     * those that have been removed manually and anonymous users who have logged
+     * out. This list is guaranteed to be in the order that the users were added
+     * to the application.
+     *
+     * - parameters:
+     *     - completionHandler: The completion handler to call when the switch is complete.
+     *                          This handler is executed on a non-main global `DispatchQueue`.
+     */
+    public func listUsers(_ completionHandler: @escaping (StitchResult<[StitchUserImpl]>) -> Void) {
+        dispatcher.run(withCompletionHandler: completionHandler) {
+            return self.listUsersInternal()()
```

Comment:
[majors]
- There are two parentheses here. I'm surprised this compiles.
- Return `[StitchUser]`. The `Impl` is an internal class only.
- This needs to be integration tested